app/views/admin/products_v3/clone.turbo_stream.haml

```diff
@@ -0,0 +1,11 @@
+- product_body = nil
+- form_with url: bulk_update_admin_products_path, method: :post, id: "products-form",
+            builder: BulkFormBuilder,
+            html: { data: { 'turbo-frame': "_self",
+              controller: "bulk-form",
+              'bulk-form-disable-selector-value': "#sort,#filters",
+              'bulk-form-error-value': defined?(@error_counts),
+              } } do |form|
+    - product_body = render partial: 'product_variant_row', locals: { form:, product: @cloned_product, product_index: @product_index, producer_options: @producer_options, category_options: @category_options, tax_category_options: @tax_category_options }, formats: :html
+= turbo_stream.before dom_id(@product) do
+    = product_body
```

Comment:
That's clever to save the rendered partial without the form around it.

Reply:
> If I remember rightly, this was because of the weird repeating issue. It would be good to have a code comment here to explain why we do it this way.

Yes, @dacook. However, this was the issue when we were passing the controller's `view_context` to create the form and then use/pass it in the partials. That was also kind of violating the view controller separations of concerns.
Now we are creating the form in the view itself, hence that repeating issue was fixed. 

**Just some more context here:**

We are using `-` for the form because we don't want the form element here. We are using the view's helper, creating a form object, and passing it to the partial in the view.
This was the major challenge/takeaway from this: avoid using the controller's `view_context` to create the view's elements and pass them in views to use. Just create a wrapper view, create that view thing you wanted from the controller, and that should be working as expected as we are just using view for view's thing. 
The root cause might be this: 
https://www.loom.com/share/271cba54a7704572ac283535184ef78b

Please let me know if you have any questions. Thanks 🙂